content/essentials/understanding-jsx.mdx

```diff
@@ -0,0 +1,106 @@
+import { Aside } from "~/components/configurable/Aside";
+
+# Understanding JSX
+
+JSX is a syntax extension for JavaScript. 
+It allows you to write HTML-like code inside your JavaScript file which keeps your rendering logic and content in the same place. 
+This provides a concise and readable way to create and represent components.
+
+## How Solid uses JSX
+
+Solid was designed to align closely with HTML standards. 
+It offers a distinct advantage: copy/paste solutions from resources like Stack Overflow, and direct usage of templates from design tools.
+
+What sets Solid's apart is the ability to return JSX. 
+This adds dynamic expressions that allow you to reference variables and functions within your HTML by using the `{ }` syntax:
+
+```jsx
+const animal = { breed: "cat", name: "Midnight" };
+const element = <p>I have a {animal.breed} named {animal.name}!</p>;
+```
+
+This allows for the injection of dynamic content into the rendered output based on the application's state or logic.
+
+Additionally, Solid's [reactive](/essentials/intro-to-reactivity) system introduces [fine-grained reactivity](/advanced/understanding-fine-grained-reactivity) with JSX.
+This ensures that only the necessary portions of the [DOM](https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Introduction) are updated when changes occur in the underlying state, optimizing efficiency and overall performance.
+
+## Using JSX in Solid
+
+### 1. Return a single root element
+
+Unlike HTML, where you can have disconnected tags at the top level, JSX requires that a component's render method returns a single root element.
+
+<Aside type="advanced">
+**Note**: When working with JSX, the system translates parts of your code into structured HTML that's placed at the start of the file.
+Static elements, which don't change, are processed differently from dynamic ones, which might change based on data or user actions. 
+For dynamic elements, special markers are added for better handling during rendering.
+<br/>
+This structure necessitates a single root element because JSX, unlike raw HTML, translates into a hierarchical object representation. 
```

Comment:
```suggestion
```

Yes, solid compiles to only a single root node, but this is exactly like HTML, and Solid supports fragments which act like multiple nodes. If you want multiple nodes in separate parts of the page, it is pretty safe to just `render` two different nodes (state gets a bit more complicated, but how to abstract state in Solid is probably a separate docs page)

Reply:
How do you propose changing this to mention that? Or do you think it's fine as is?